integrationtests/src/test/java/org/axonframework/integrationtests/queryhandling/QueryThreadingIntegrationTest.java

```diff
@@ -0,0 +1,193 @@
+            new AxonServerContainer()
+                    .withAxonServerName("axonserver")
+                    .withAxonServerHostname(HOSTNAME)
+                    .withDevMode(true)
+                    .withEnv("AXONIQ_AXONSERVER_INSTRUCTION-CACHE-TIMEOUT", "1000")
+                    .withImagePullPolicy(PullPolicy.ageBased(Duration.ofDays(1)))
+                    .withNetworkAliases("axonserver");
+
+    private AxonServerConnectionManager connectionManager;
+    private AxonServerQueryBus queryBus;
+    private AxonServerQueryBus queryBus2;
+
+    @BeforeEach
+    void setUp() {
+        Serializer serializer = JacksonSerializer.defaultSerializer();
+
+        String server = axonServer.getHost() + ":" + axonServer.getGrpcPort();
+        AxonServerConfiguration configuration = AxonServerConfiguration.builder()
+                .componentName("threadingTest")
+                .servers(server)
+                .build();
+        configuration.setCommandThreads(5);
+        configuration.setQueryThreads(5);
+        configuration.setQueryResponseThreads(5);
+        connectionManager = AxonServerConnectionManager.builder()
+                .axonServerConfiguration(configuration)
+                .channelCustomizer(ManagedChannelBuilder::directExecutor)
+                .build();
+        connectionManager.start();
+
+
+        // The application having a query that depends on another one
+        QueryBus localQueryBus = SimpleQueryBus.builder().build();
+        queryBus = AxonServerQueryBus.builder()
+                .axonServerConnectionManager(connectionManager)
+                .configuration(configuration)
+                .localSegment(localQueryBus)
+                .updateEmitter(localQueryBus.queryUpdateEmitter())
+                .messageSerializer(serializer)
+                .genericSerializer(serializer)
+                .build();
+        queryBus.start();
+
+        // The secondary application
+        QueryBus localQueryBus2 = SimpleQueryBus.builder().build();
+        queryBus2 = AxonServerQueryBus.builder()
+                .axonServerConnectionManager(connectionManager)
+                .configuration(configuration)
+                .localSegment(localQueryBus2)
+                .updateEmitter(localQueryBus.queryUpdateEmitter())
+                .messageSerializer(serializer)
+                .genericSerializer(serializer)
+                .build();
+        queryBus2.start();
+        waitingQueries.set(0);
+    }
+
+    @AfterEach
+    void tearDown() {
+        queryBus.shutdownDispatching();
+        queryBus.disconnect();
+        queryBus2.shutdownDispatching();
+        queryBus2.disconnect();
+
+        connectionManager.shutdown();
+    }
+
+    @SuppressWarnings("resource")
+    @Test
+    void canStillHandleQueryResponsesWhileManyQueriesHandling() throws InterruptedException {
+        queryBus2.subscribe("query-b", String.class, query -> {
+            log.info("Query B received");
```

Comment:
I think we can remove the logging, right? I assume you used that to figure out if the process went as intended.